src/dc_receive_imf.rs

```diff
@@ -5116,4 +5121,70 @@ Reply from different address
 
         Ok(())
     }
+
+    #[async_std::test]
+    async fn test_no_private_reply_to_blocked_account() -> Result<()> {
+        let mut tcm = TestContextManager::new().await;
+        let alice = tcm.alice().await;
+        let bob = tcm.bob().await;
+
+        // =============== Bob creates a group ===============
+        let group_id =
+            chat::create_group_chat(&bob, ProtectionStatus::Unprotected, "Group").await?;
+        chat::add_to_chat_contacts_table(
+            &bob,
+            group_id,
+            bob.add_or_lookup_contact(&alice).await.id,
+        )
+        .await?;
+
+        // =============== Bob sends the first message to the group ===============
+        let sent = bob.send_text(group_id, "Hello all!").await;
+        alice.recv_msg(&sent).await;
+
+        let chats = Chatlist::try_load(&bob, 0, None, None).await?;
+        assert_eq!(chats.len(), 1);
+
+        // =============== Bob blocks Alice ================
+        Contact::block(&bob, bob.add_or_lookup_contact(&alice).await.id).await?;
+
+        // =============== Alice replies private to Bob ==============
+        let received = alice.get_last_msg().await;
+        assert_eq!(received.text, Some("Hello all!".to_string()));
+
+        let received_group = Chat::load_from_db(&alice, received.chat_id).await?;
+        assert_eq!(received_group.typ, Chattype::Group);
+
+        let mut msg_out = Message::new(Viewtype::Text);
+        msg_out.set_text(Some("Private reply".to_string()));
+        msg_out.set_quote(&alice, Some(&received)).await?;
+
+        let alice_bob_chat = alice.create_chat(&bob).await;
+        let sent2 = alice.send_msg(alice_bob_chat.id, &mut msg_out).await;
+        bob.recv_msg(&sent2).await;
+
+        // ========= check that no contact request was created ============
+        let chats = Chatlist::try_load(&bob, 0, None, None).await.unwrap();
+        assert_eq!(chats.len(), 1);
+        let chat_id = chats.get_chat_id(0).unwrap();
+        let chat = Chat::load_from_db(&bob, chat_id).await.unwrap();
+
+        // since only chat is a group, no new open chat has been created
+        assert_eq!(chat.typ, Chattype::Group);
+        let received = bob.get_last_msg().await;
+        assert_eq!(received.text, Some("Hello all!".to_string()));
+
+        // =============== Bob unblocks Alice ================
+        // test if the blocked chat is restored correctly
+        Contact::unblock(&bob, bob.add_or_lookup_contact(&alice).await.id).await?;
+        let chats = Chatlist::try_load(&bob, 0, None, None).await.unwrap();
+        assert_eq!(chats.len(), 2);
+        let chat_id = chats.get_chat_id(0).unwrap();
+        let chat = Chat::load_from_db(&bob, chat_id).await.unwrap();
+        assert_eq!(chat.typ, Chattype::Single);
+        let received = bob.get_last_msg().await;
+        assert_eq!(received.text, Some("Private reply".to_string()));
```

Comment:
**question for core devs:** does this mean that currently if blocked contacts send images and other files they are also received/processed as usual, just hidden and when you unblock the contact all the spam will be there? (ex. if that person sends you a lot of messages with images etc) 

I was thinking the behavior was to completely ignore those messages

Reply:
Looks like it, but I am very much not the person you can talk this through with - maybe just open another issue

Reply:
Yes, seems "blocked chat" is a hidden chat that can reappear later on and takes space in the database. Maybe chats should be preserved, but messages trashed immediately or something like this.

Blocking is rarely used and needs a thorough UX design, especially wrt. multi-device.